plugins/catalog-backend/src/next/database/DefaultProcessingDatabase.test.ts

```diff
@@ -23,33 +23,187 @@ import {
+        kind: 'Location',
+        metadata: {
+          name: 'fakelocation',
+        },
+        spec: {
+          type: 'url',
+          target: 'somethingelse',
+        },
+      };
+    });
+
+    it('fails when there is no processing state for the entity', async () => {
+      await processingDatabase.transaction(async tx => {
+        await expect(() =>
+          processingDatabase.updateProcessedEntity(tx, {
+            id,
+            processedEntity,
+            state: new Map<string, JsonObject>(),
+            relations: [],
+            deferredEntities: [],
+          }),
+        ).rejects.toThrow(`Processing state not found for ${id}`);
+      });
+    });
+
+    it('updates the refresh state entry with the cache, processed entity and errors', async () => {
+      await insertRefreshStateRow({
+        entity_id: id,
+        entity_ref: 'location:default/fakelocation',
+        unprocessed_entity: '{}',
+        processed_entity: '{}',
+        errors: '',
+        next_update_at: 'now()',
+        last_discovery_at: 'now()',
+      });
+
+      const state = new Map<string, JsonObject>();
+      state.set('hello', { t: 'something' });
+
+      await processingDatabase.transaction(tx =>
+        processingDatabase.updateProcessedEntity(tx, {
+          id,
+          processedEntity,
+          state,
+          relations: [],
+          deferredEntities: [],
+          errors: 'something broke',
+        }),
       );
-    };
 
-    const insertRefreshStateRow = async (ref: DbRefreshStateRow) => {
-      await db<DbRefreshStateRow>('refresh_state').insert(ref);
-    };
+      const [entity] = await db<DbRefreshStateRow>('refresh_state').select();
+
+      expect(entity.processed_entity).toEqual(JSON.stringify(processedEntity));
+      expect(entity.cache).toEqual(JSON.stringify(state));
+      expect(entity.errors).toEqual('something broke');
+    });
+
+    it('removes old relations and stores the new relationships', async () => {
+      await insertRefreshStateRow({
+        entity_id: id,
+        entity_ref: 'location:default/fakelocation',
+        unprocessed_entity: '{}',
+        processed_entity: '{}',
+        errors: '',
+        next_update_at: 'now()',
+        last_discovery_at: 'now()',
+      });
+
+      const relations = [
+        {
+          source: {
+            kind: 'Component',
+            namespace: 'Default',
+            name: 'foo',
+          },
+          target: {
+            kind: 'Component',
+            namespace: 'Default',
+            name: 'foo',
+          },
+          type: 'memberOf',
+        },
+      ];
+
+      await processingDatabase.transaction(tx =>
+        processingDatabase.updateProcessedEntity(tx, {
+          id,
+          processedEntity,
+          state: new Map<string, JsonObject>(),
+          relations: relations,
+          deferredEntities: [],
+        }),
+      );
+
+      const [savedRelations] = await db<DbRelationsRow>('relations')
```

Comment:
same here, best to get all of them as an array and assert that you got exactly one row back instead